T&A 45956: Fixes withKioskMode type error

matheuszych opened the pull request:
https://mantis.ilias.de/view.php?id=45956

Aims to fix withKioskMode type error.

Already reviewed by @thojou.

matheuszych committed:
Fixes withKioskMode type error

kergomard commented:
Thank you very much @matheuszych for the PR!

One question though: Why do we need this cast here: https://github.com/ILIAS-eLearning/ILIAS/pull/10348/files#diff-a293cd08c2c060fd8ed019c010e376e3314cbd4c65f85d2e54fbd4488176937eR639 ?

I would very much love to get rid of it as we should always have `int`s here and keep the fix fully localized to applying the defaults.

I assume this will then need to be picked to ILIAS 10, but not 11 and above, right?

Best,
@kergomard

matheuszych commented:
Hello @kergomard,

i was not able to reproduce the error because of which i added a typecast there. I removed it for now.

I also tested the behavior in ILIAS 10 and it seems to be broken there aswell, thus this will also need to be picked to the higher versions.

Do you want me to provide PRs for that?

Best regards
@matheuszych

kergomard commented:
Thank you very much @matheuszych !

I merged and picked this to ILIAS 10. In 11 the Templates were completely rewritten, so I assume this type of error cannot happen anymore, but maybe you would like to have a look?

Best,
@kergomard

matheuszych commented:
Hello @kergomard,

i checked it in ILIAS 11 and it indeed seems to not be possible anymore.

Best regards
@matheuszych
